submissions/SamVal007/Memory_Pair_Game/main.js

```diff
@@ -0,0 +1,134 @@
+    flipped: 'flipped',
+    hidden: 'hidden'
+};
+
+const cleanCountArray = () => similarCards = [];
+
+const flippSelectedCards = () => {
+    setTimeout(() => {
+        document.querySelectorAll(".flipped").forEach((card) => {
+            card.classList.remove(mainAttributes.flipped);
+            cleanCountArray();
+            blockedClick = false;
+        });
+    }, 900);
+};
+
+const hideSimilarCards = () => {
+    setTimeout(() => {
+        document.querySelectorAll(".flipped").forEach((card) => {
+            card.classList.add(mainAttributes.hidden);
+            cleanCountArray();
+            blockedClick = false;
+            const counterHiddenCards = document.querySelectorAll(".hidden").length;
+            const amountOfCards = (images.length * 2);
+            console.log(amountOfCards);
+            if (counterHiddenCards == amountOfCards) {
+                victoryMessage();
+            }
+        });
+    }, 600);
+};
+
+const compareSimilarCards = () => {
+    const countCards = similarCards.length;
+    if (countCards == 2) {
+        blockedClick = true;
+        if (similarCards[0] === similarCards[1]) {
+            hideSimilarCards();
+        } else {
+            flippSelectedCards();
+        }
+    }
+};
+
+const selectCard = ({
+    target
+}) => {
+    if (!blockedClick) {
+        const selectedCard = target.closest('.card');
+        if (selectedCard && !selectedCard.classList.contains('flipped')) {
+            let alt = target.parentNode.nextSibling.firstChild.getAttribute("alt");
+            selectedCard.classList.add(mainAttributes.flipped);
+            similarCards.push(alt);
+            compareSimilarCards();
+        }
+    }
+};
+
+const drawCards = () => {
+    const fragment = [];
+
+    const game = document.createElement('div');
+    game.classList.add(mainAttributes.game);
+
+    const arrImages = [...images, ...images];
+    const shuffleArray = arrImages.sort(function () {
+        return 0.5 - Math.random()
+    });
+
+    shuffleArray.forEach((item) => {
+        const card = document.createElement('div');
+        const frontCard = document.createElement('div');
+        const backCard = document.createElement('div');
+        const frontImg = document.createElement('img');
+        const backImg = document.createElement('img');
+
+        card.classList.add(mainAttributes.card);
+        frontCard.classList.add(mainAttributes.front);
+        backCard.classList.add(mainAttributes.back);
+        frontImg.classList.add(mainAttributes.frontImg);
+        frontImg.src = `img/${item}.png`;
+        frontImg.alt = `${item}`;
+        backImg.classList.add(mainAttributes.backImg);
+        backImg.src = 'img/back_logo.png';
+        backImg.alt = 'logo_card';
+
+        frontCard.append(frontImg);
+        backCard.append(backImg);
+        card.append(backCard);
+        card.append(frontCard);
+        game.append(card);
+    });
+    fragment.push(game);
+
+    return fragment;
```

Comment:
why just not return game ? looks like fragment variable is an over usage

Reply:
Yep! Very Good point!
Thanks!